pages/posts/page/[page_number].tsx

```diff
@@ -0,0 +1,87 @@
+import fs from 'node:fs';
+import type { GetStaticPaths, GetStaticProps } from 'next';
+import type { DualPost } from '../../../lib/types';
+import PostPreview from '../../../components/PostPreview';
+import { getAllPosts } from '../../../lib/getAllPosts';
+import { podcastXml } from '../../../lib/podcast';
+import PageWrapper from '../../../components/PageWrapper';
+import Paginator from '../../../components/Paginator';
+import { paginate } from '../../../lib/helpers';
+
+export const getStaticProps: GetStaticProps = async (context) => {
+  const allPosts = await getAllPosts();
+  const whichPage = Number(context.params?.page_number);
+  const thisPagePosts = paginate(
+    allPosts.sort(
+      (a, b) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime(),
+    ),
+    whichPage,
+    8,
+  );
+  const enPosts = allPosts.map((dual) => dual.en);
+  const esPosts = allPosts.map((dual) => dual.es);
+  fs.writeFileSync(`./public/podcast.en.rss`, podcastXml(enPosts));
+  fs.writeFileSync(`./public/podcast.es.rss`, podcastXml(esPosts));
```

Comment:
i think these will need to be relocated.  probably could be punted to an issue for a later PR, because it's not really hurting that much, but as the site grows over time, this could slow down the build time.  that's because this will generate the podcasts once for every page of the pagination.  

if you console.log in here and run a build locally, you should see 3 generations of the podcast.

Reply:
we should explore only querying cosmic for the posts we need, instead of fetching all of them and slicing.  not crucial or anything, but should be doable, and possibly speed up build times and decrease api usage. i almost guarantee that cosmic has some way of passing `?limit=8&offset=16` or something like that, so that we can be more surgical about what we fetch.  just spitballing, maybe worth opening an issue, but doesn't need to be done for this PR, obviously.